drivers/clock_mgmt/Kconfig

```diff
@@ -0,0 +1,66 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+#
+# Clock controller drivers
+#
+menuconfig CLOCK_MGMT
```

Comment:
Yes, just an abbreviation. I chose MGMT because it is (as far as I know) a common abbreviation for MANAGEMENT

Reply:
Can you change to management? `mgmt` is what MCUmgr uses and this is confusing

Reply:
IE `CONFIG_CLOCK_MANAGEMENT`? Sure, it will just result in some of the Kconfig names being relatively long

Reply:
I agree. `mgmt` is a quite frequent short name. It can't be hogged by MCUmgr.

Note also that in the tree we can find already:
NET_MGMT, WIFI_MGMT, SETTING_MGMT, SHELL_MGMT, STAT_MGMT, OS_MGMT, ....

Reply:
> Note also that in the tree we can find already:
> NET_MGMT, WIFI_MGMT, SETTING_MGMT, SHELL_MGMT, STAT_MGMT, OS_MGMT, ....

4 of the ones you listed are MCUmgr.

I would suggest clock_management, we do not shorten words in filenames

Reply:
> 4 of the ones you listed are MCUmgr.

Sure, but two aren't, and I doubt anyone is expecting `NET_MGMT` to be related to MCUmgr. Honestly as I consider it more I'd really prefer to use the abbreviated name. The macros for this subsystem are already rather long, and I doubt anyone is going to expect `CLOCK_MGMT` is related to MCUmgr functionality.

> I would suggest clock_management, we do not shorten words in filenames (note: filename update is the main point)

Not sure I follow here- are you suggesting we expand filenames to `clock_management` but keep macros and Kconfigs as `CLOCK_MGMT`?

Reply:
Macros and filenames should be clock_management/CLOCK_MANAGEMENT, I don't buy adding 6 letters to make a word proper makes it "too long"

Reply:
I'd prefer shorter macro expansions, also for consistency sakes.

Reply:
Consistency with what? Macros in zephyr are not shortened

Reply:
> Consistency with what? Macros in zephyr are not shortened
- `DT_PINCTRL_BY_IDX`: DEVICETREE_PIN_CONTROL_BY_INDEX
- `SYS_INIT`: SYSTEM_INITIALIZATION
- `DMIC_STATE_UNINIT`: DMIC_STATE_UNINITIALIZED

I know we also often expand macros to their full name in Zephyr, but we definitely shorten macros as well. Since these macros will already be rather long I'd really rather shorten them a bit to make formatting easier in driver files.

Reply:
SYS_INIT comes from sysVinit on linux. DT is a common acronym for device tree, it's even used in talks "...how we do things in D T...". Uninit is also a common thing. No-one shortens management to mgmt, all you can find on a search engine is a band (unrelated but I dislike the shortening in MCUmgr, there is no reason it can't be image management, img mgmt just looks lazy, but that's how the original creator named it)

Reply:
`CONFIG_CACHE_MANAGEMENT`, `CONFIG_NRF_GRTC_TIMER_CLOCK_MANAGEMENT`, `CONFIG_NET_POWER_MANAGEMENT`

Reply:
> No-one shortens management to mgmt, all you can find on a search engine is a band

If I stick `clk` in google, I don't get results about clocks, I hear about a Mercedes-Benz. Googling "what is mgmt short for" yields the result you'd expect. Regardless, it is clear we do frequently expand MGMT out to MANAGEMENT in Zephyr, and I don't feel too strongly about this. I'll replace instances of `MGMT` with `MANAGEMENT` in the next push of this PR

Reply:
Thanks!

Reply:
Updated. All macros, folder names, Kconfigs, and file names that used `clock_mgmt` or `CLOCK_MGMT` are now replaced with `CLOCK_MANAGEMENT`